### ISSUE TITLE
Sylius messenger configuration won't work with AMQP transport

### ISSUE THREAD

**Issue by andriusvo** [Sylius/Sylius#14670]
**Sylius version affected**: 1.12

**Description**  
Messenger transports can only be used within Doctrine transport, because of configuration `queue_name`.
There is a configuration of messenger defined in:
`Sylius/Bundle/CoreBundle/Resources/config/app/messenger.yaml`

And you see here `queue_name` options. According to the documentation and code, `queue_name` is only available for Doctrine transports and if I want to use AMQP for example, so I won't be able to do that.

Docs: https://symfony.com/doc/current/messenger.html#prioritized-transports

```
# queue_name is specific to the doctrine transport
queue_name: high
```

**Steps to reproduce**  
---

**Possible Solution**  
---

**Comment by Roshyo** [comment]
I confirm this issue and can reproduce.
@andriusvo do you have any possible solution in mind?

I will try to think of it also.

**Comment by andriusvo** [comment]
The first idea that came to my mind is to load different configuration files based on usable transport in the project.

**Comment by Rafikooo** [comment]
Hello Comrades, I'd like to share with you what I've learned. I've played with `AMQP` and `RabbitMQ` for the first time, installed the necessary PHP extension, created an account, and switched the `MESSENGER_TRANSPORT_DSN` connection string to `AMQP` and everything we have configured in Sylius worked right out of the box. 
My thoughts:
1. `queue_name` is indeed dedicated only to Doctrine transport, but doesn't conflict with the `AMQP`
2. Symfony sets an initial state when used with `AMQP` - described [here](https://symfony.com/doc/current/messenger.html#amqp-transport)
3. by default, it creates a single `messages` exchange with a single queue that handles all traffic, both catalog recalculation (a high priority in Sylius case) and catalog promotion removal (normal priority).
4. We can somehow unify a configuration to use the same names but since `RabbitMQ` has an exchange (which Doctrine transport hasn't) it may depend on the needs of the developer how the `AMQP` transport will be configured

As I said at first glance it works but maybe I missed something, feel free to expand on this discussion
